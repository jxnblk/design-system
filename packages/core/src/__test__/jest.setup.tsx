import React from 'react'
import { ThemeProvider } from 'styled-components'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import { createTheme } from '..'

global.rendererCreateWithTheme = (tree) =>
  renderer.create(<ThemeProvider theme={createTheme()}>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
