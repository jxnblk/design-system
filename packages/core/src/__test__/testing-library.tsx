import { render, RenderResult } from '@testing-library/react'
import PropTypes from 'prop-types'
import React from 'react'

import { ThemeProvider } from '..'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
