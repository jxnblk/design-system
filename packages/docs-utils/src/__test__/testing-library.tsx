import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

// @ts-ignore
const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
