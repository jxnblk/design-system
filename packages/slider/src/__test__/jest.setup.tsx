import React from 'react'
import { ThemeProvider } from 'pcln-design-system'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

global.rendererCreateWithTheme = (tree) => renderer.create(<ThemeProvider>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
