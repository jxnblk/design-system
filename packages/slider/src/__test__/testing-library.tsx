import type { RenderOptions, RenderResult } from '@testing-library/react'
import type { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
