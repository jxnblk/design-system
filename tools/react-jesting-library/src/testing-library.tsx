import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

/**
 * customRender wraps the render method of \@testing-library/react and wraps it in our
 * custom AllTheProviders which provides a pcln-design-system theme
 *
 * @public
 */
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
