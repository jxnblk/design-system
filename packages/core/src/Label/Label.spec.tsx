import React from 'react'
import { fireEvent, render, screen } from '../__test__/testing-library'
import { Label } from '..'

describe('Label', () => {
  test('it renders', () => {
    const json = rendererCreateWithTheme(<Label />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Label hidden renders', () => {
    const json = rendererCreateWithTheme(<Label hidden />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Label nowrap renders', () => {
    const json = rendererCreateWithTheme(<Label nowrap />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Label width renders', () => {
    const json = rendererCreateWithTheme(<Label width={1 / 2} nowrap />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Label clickable renders with cursor pointer', () => {
    const mockOnClick = jest.fn()
    render(<Label onClick={mockOnClick}>Clickable Label</Label>)

    const label = screen.getByText('Clickable Label')
    expect(label).toHaveStyleRule('cursor', 'pointer')

    expect(mockOnClick).toHaveBeenCalledTimes(0)
    fireEvent.click(label)
    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
})
