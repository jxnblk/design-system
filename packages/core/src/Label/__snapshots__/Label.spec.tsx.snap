// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Label hidden renders 1`] = `
.c0 {
  display: block;
  width: 100%;
  margin: 0;
  color: #4f6f8f;
  font-size: 10px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  position: absolute;
  width: 1px;
  height: 1px;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

<label
  className="c0"
  color="text.light"
  fontSize="10px"
  fontWeight={700}
  hidden={true}
  letterSpacing="0.2px"
/>
`;

exports[`Label Label nowrap renders 1`] = `
.c0 {
  display: block;
  width: 100%;
  margin: 0;
  color: #4f6f8f;
  font-size: 10px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  white-space: nowrap;
}

<label
  className="c0"
  color="text.light"
  fontSize="10px"
  fontWeight={700}
  letterSpacing="0.2px"
/>
`;

exports[`Label Label width renders 1`] = `
.c0 {
  display: block;
  width: 100%;
  margin: 0;
  color: #4f6f8f;
  font-size: 10px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  width: 50%;
  white-space: nowrap;
}

<label
  className="c0"
  color="text.light"
  fontSize="10px"
  fontWeight={700}
  letterSpacing="0.2px"
  width={0.5}
/>
`;

exports[`Label it renders 1`] = `
.c0 {
  display: block;
  width: 100%;
  margin: 0;
  color: #4f6f8f;
  font-size: 10px;
  font-weight: 700;
  line-height: 1.4;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

<label
  className="c0"
  color="text.light"
  fontSize="10px"
  fontWeight={700}
  letterSpacing="0.2px"
/>
`;
